### PR TITLE
[ext] New Getting Started tutorial (extension service worker)

### PR DESCRIPTION
**Pull request by AmySteam**
## Scope

- Adds a new tutorial to the Getting Started guides called "Quick API reference"

## TODO:

- [x] Finish Test that it works / Potential enhancements
- [x] Add links to the new explainer.
- [x] Merge code sample [PR#851](https://github.com/GoogleChrome/chrome-extensions-samples/pull/851)

### CODE REVIEW

**Review by sebastianbenz — COMMENTED** [review]
Looking good, left a few in between meetings comments. Will continue later.

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+
+{% Aside %}
+
+Remember to set `type.module` when using a modern module bundler framework.
+
+{% endAside %}
+
+### _Optional: Debugging the service worker_ {: #step-3 }
+
+Let's quickly go over how to find the service worker logs and know when it has terminated. Follow the instructions to [Load an unpacked extension][doc-dev-basics-unpacked] and wait 30 seconds for the service worker to stop. Click on the "service worker" hyperlink to inspect it.
```
[the remark this answers is not in the document]
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:115]: Good suggestion! 👍

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "permissions": ["storage"],
+}
+```
+
+The following code save the default suggestions to storage when the extension is first installed:
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:148]:
```suggestion
The following code save the default suggestions to storage when the extension is first installed by listening to the `onInstalled` event:
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+      apiSuggestions: ['tabs', 'storage', 'scripting']
+    });
+  }
+});
+```
+
+Service workers do not have direct access to the [window object][mdn-window], therefore cannot use
+[window.localStorage()][mdn-local-storage] to store values. Also, service workers are short-lived execution environments;
+they get terminated repeatedly throughout a user's browser session, which makes it incompatible with
+global variables.
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:167]:
```suggestion
global variables. Instead, we use `chrome.storage.local` which stores data on the local machine.
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+
+See [Saving state](TBD) to learn about storage options for extension service workers.
+
+### Step 5: Register your events {: #step-5 }
+
+All event listeners need to be registered in the global scope of the service worker. In other words, event listeners should not be nested in functions. This way Chrome can immediately invoke all event handlers, even if the extension's async startup logic hasn't finished.
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:173]: Not sure if this is a better way to phrase this:

```suggestion
All event listeners need to be statically registered in the global scope of the service worker. In other words, event listeners should not be nested in async functions. This way Chrome can ensure that all event handlers are restored in case of a service worker reboot.
```

Maybe rephrase to:

"

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+  chrome.tabs.create({ url: URL_CHROME_EXTENSIONS_DOC + input });
+  // Save the latest keyword
+  updateHistory(input);
+});
+
+async function updateHistory(input) {
+  const { apiSuggestions } = await chrome.storage.local.get('apiSuggestions');
+  apiSuggestions.unshift(input);
+  apiSuggestions.splice(NUMBER_OF_PREVIOUS_SEARCHES);
+  await chrome.storage.local.set({ apiSuggestions });
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:227]: Easier way (I think):

```
const { apiSuggestions } = await chrome.storage.local.get('apiSuggestions');
apiSuggestions = input.concat(apiSuggestions);
await chrome.storage.local.set({ apiSuggestions });
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 6]
```diff
@@ -0,0 +1,469 @@
...
+  },
+}
+```
+
+{% Aside 'important' %}
+The [`"minimum_chrome_version"`][manifest-min-version] explains how this key behaves when a user tries to install your extension but isn't using a compatible version of Chrome.
+
+{% endAside %}
+
+The following code registers the omnibox event listeners at the top level of the script and updates [chrome.storage][api-storage] with the most recent api search.
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:194]: The code also creates suggestions. Maybe add a bit more explanation and maybe even split up the code.

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 7]
```diff
@@ -0,0 +1,469 @@
...
+
+{% endAside %}
+
+### Step 6: Set up a recurring event {: #step-6 }
+
+The `setTimeout()` or `setInterval()` methods are commonly used to perform delayed or periodic
+tasks. However, these APIs can fail because the scheduler will cancel the timers when the service
+worker is terminated. Instead, extensions can use the [`chrome.alarms`][api-alarms] API. 
+
+To use the Alarms API, request the `"alarms"` permission in the manifest. The extension also needs to request [host permission][doc-host-perm] to retrieve the extension tips from the glitch site:
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:244]:
```suggestion
To use the Alarms API, request the `"alarms"` permission in the manifest. The extension also needs to request [host permission][doc-host-perm] to be able to fetch the extension tips from a remote hosted location:
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 8]
```diff
@@ -0,0 +1,469 @@
...
+
+{% Label %}sw-tips.js:{% endLabel %}
+
+```js
+// Fetch tip & save in storage
+const updateTip = async () => {
+  const response = await fetch('https://extension-tips.glitch.me/tips.json');
+  const tips = await response.json();
+  const randomIndex = Math.floor(Math.random() * tips.length);
+  await chrome.storage.local.set({ tip: tips[randomIndex] });
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:268]: remove the await here, but return the promise from set

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 9]
```diff
@@ -0,0 +1,469 @@
...
+```json/2/3
+{
+  ...
+  "permissions": ["storage", "alarms"],
+  "permissions": ["storage"],
+  "host_permissions": ["https://extension-tips.glitch.me/*"],
+}
+```
+
+The following code sets up an alarm once a day to fetch the daily tip and save it to
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:257]: Maybe mention that you create the alarm at install time

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 10]
```diff
@@ -0,0 +1,469 @@
...
+
+{% Aside %}
+
+All [Chrome API][doc-apis] event listeners and methods restart the service worker 30 second termination timer. See [Extension service worker lifecycle](tdb) to learn more.
+
+{% endAside %}
+
+### Step 7: Communicate with other contexts {: #step-7 }
+
+[Content scripts][doc-content] communicate with the rest of the extension through [message passing][doc-messages]. In this example, the content script will request the tip of the day from the service worker.
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:292]:
```suggestion
[Content scripts][doc-content] communicate with the rest of the extension through [message passing][doc-messages]. In this example, the content script will request the tip of the day from the service worker when the user browses the Chrome extension documentation. 
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 11]
```diff
@@ -0,0 +1,469 @@
...
+    {
+      "matches": ["https://developer.chrome.com/docs/extensions/reference/*"],
+      "js": ["content.js"]
+    }
+  ]
+}
+
+```
+
+Create a new content file. The following code generates a button that will open the tip popover. It also sends a message to the service worker requesting the extension tip.
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:311]: Maybe mention that content scripts can't access storage.local.

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 12]
```diff
@@ -0,0 +1,469 @@
...
+
+TBD
+
+- DomParser
+- Popover API
+- SVG element
+
+{% endDetails %}
+
+The following code sends the daily tip from the service worker to the content script.
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:360]:
```suggestion
The final step is to add a message handler to our service worker that replies the daily tip when requested from the content script. 
```

**Review by sebastianbenz — COMMENTED** [review]
Few more comments, only small things. Overall I think this is great!

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md (hunk 5 above)
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:227]: concat behaves funny 🙃
Screenshot 2023-03-22 at 09 19 31

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 13]
```diff
@@ -0,0 +1,445 @@
+---
+layout: 'layouts/doc-post.njk'
+title: 'Quick API Reference'
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:3]: I still think we should use a more descriptive name here. Afaik, title will be used in the article listings and it'd be great to be able to tell from the title what this tutorial. is about (please correct me if I'm wrong).

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md (hunk 5 above)
Comment (sebastianbenz) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:228]: Sorry, should've been `[input].concat`. Thinking. more about this, I would ignore the number of suggestions and just go with:

```
const { apiSuggestions } = await chrome.storage.local.get('apiSuggestions');
await chrome.storage.local.set({ [input].concat(apiSuggestions) });
```

But your call :-)

Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:228]: Okie dokies 😁

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:3]: I'm making the change here and also changing the rest of the Getting Started guide tutorials to match the new approach. Thanks for your suggestion 👍

[the remark this answers is not in the document]
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:3]: [PR#5769](https://github.com/GoogleChrome/developer.chrome.com/pull/5769)

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 14]
```diff
@@ -0,0 +1,445 @@
...
+## Overview {: #overview }
+
+This tutorial builds an extension that allows users to open Chrome API reference pages using the omnibox. It also provides a daily Chrome extension tip.
+
+{% Video src="video/BhuKGJaIeLNPW9ehns59NfwqKxF2/WmVEGpEZ9ts1J0pUOzEr.mp4", width="600", height="398", autoplay="true", muted="true"%}
+
+This article will explain how to do the following tasks in an extension service worker:
+
+- Register a service worker and import modules.
+- Find extension service worker logs.
```
[the remark this answers is not in the document]
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md]: I rewrote the list 👍

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 15]
```diff
@@ -0,0 +1,445 @@
...
+    "service_worker": "service-worker.js",
+  },
+}
+```
+
+Extensions register their service worker in the manifest, which only takes a single JavaScript file.
+There's no need to use `navigator.serviceWorker.register()`, like you would in a web app. See
+[Differences between extension and web service workers](tbd) to learn more.
+
+You can download the icons located on the [Github repo][github-open-api].
```
[the remark this answers is not in the document]
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:58]: Done 👍

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md (hunk 3 above)
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:170]: Added this on a separate commit 👍

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md (hunk 6 above)
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:197]: Good idea. On it! 🏃🏻‍♀️💨

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md (hunk 8 above)
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:271]: I'm going to go through and update the code to match the sample repo. On it! 🏃🏻‍♀️💨

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md (hunk 11 above)
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:336]: Technically, they DO have access to the storage API, but for security reasons, we are encouraged to use messages instead.

**Comment by AmySteam** [comment]
@sebastianbenz Thanks for sharing your insights! 💯 This is ready for another round of tech review 😄

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 16]
```diff
@@ -0,0 +1,507 @@
...
+
+See [Debugging extensions](tbd) for more ways debug the extension service worker.
+
+{% Aside 'caution' %}
+Don't forget to fix the file name before moving on!
+{% endAside %}
+
+### Step 4: Initialize the state {: #step-4 }
+
+Extensions can save initial values to storage. To use the [`chrome.storage`][api-storage] API, we need to request permission in the manifest:
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:140]:
```suggestion
Chrome will shut down service workers if they are not needed. We use the [`chrome.storage`][api-storage] API to persist state across service worker sessions. For storage access, we need to request permission in the manifest:
```
I like to give a tiny bit of the why when starting a new step in a tutorial - wdyt?

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 17]
```diff
@@ -0,0 +1,507 @@
...
+      apiSuggestions: ['tabs', 'storage', 'scripting']
+    });
+  }
+});
+```
+
+Service workers do not have direct access to the [window object][mdn-window], therefore cannot use
+[window.localStorage()][mdn-local-storage] to store values. Also, service workers are short-lived execution environments;
+they get terminated repeatedly throughout a user's browser session, which makes it incompatible with
+global variables.  Instead, we use [`chrome.storage.local`](tbd) which stores data on the local machine.
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:170]:
```suggestion
global variables.  Instead, we use [`chrome.storage.local`](/docs/extensions/reference/storage/#property-local) which stores data on the local machine.
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 18]
```diff
@@ -0,0 +1,507 @@
...
+}
+```
+
+{% Aside 'important' %}
+The [`"minimum_chrome_version"`][manifest-min-version] explains how this key behaves when a user tries to install your extension but isn't using a compatible version of Chrome. 
+
+{% endAside %}
+
+Now let's register the omnibox event listeners at the top level of the script. The
+[`onInputChanged()`](tbd) takes the current user input and a [suggestResult][omnibox-suggest]
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:198]: Have we already got these links?

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 19]
```diff
@@ -0,0 +1,507 @@
...
+object. The keywords in storage will populate the suggestions. When user enters the omnibox keyword
+(api) in the address bar followed by tab or space, Chrome will display a list of suggestions.
+
+Now, let's register the omnibox event listeners at the top level of the script. When the user enters the omnibox keyword "api" in the address bar followed by tab or space, Chrome will display a list of suggestions based on the keywords in storage. The [onInputChanged()][omnibox-input-changed] event, which takes the current user input and a suggestResult object, is responsible for populating these suggestions.
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:202]:
```suggestion
Now, let's register the omnibox event listeners at the top level of the script. When the user enters the omnibox keyword `api` in the address bar followed by tab or space, Chrome will display a list of suggestions based on the keywords in storage. The [onInputChanged()][omnibox-input-changed] event, which takes the current user input and a suggestResult object, is responsible for populating these suggestions.
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 20]
```diff
@@ -0,0 +1,507 @@
...
+
+{% endAside %}
+
+### Step 6: Set up a recurring event {: #step-6 }
+
+The `setTimeout()` or `setInterval()` methods are commonly used to perform delayed or periodic
+tasks. However, these APIs can fail because the scheduler will cancel the timers when the service
+worker is terminated. Instead, extensions can use the [`chrome.alarms`][api-alarms] API. 
+
+Start by requesting the `"alarms"` permission in the manifest. Additionally, to fetch the extension tips from a remote hosted location, you need to request [host permission][doc-host-perm]:
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:265]:
```suggestion
Start by requesting the `alarms` permission in the manifest. Additionally, to fetch the extension tips from a remote hosted location, you need to request [host permission][doc-host-perm]:
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 21]
```diff
@@ -0,0 +1,507 @@
...
+```json/2/3
+{
+  ...
+  "permissions": ["storage", "alarms"],
+  "permissions": ["storage"],
+  "host_permissions": ["https://extension-tips.glitch.me/*"],
+}
+```
+
+The extension will fetch all the tips and pick one at random. Then save it to storage. We will create an alarm that will be triggered once a day. Alarms are not saved when you close Chrome. So we need to check if the alarm exists and create it if it doesn't.
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:278]:
```suggestion
The extension will fetch all the tips, pick one at random and save it to storage. We will create an alarm that will be triggered once a day to update the tip. Alarms are not saved when you close Chrome. So we need to check if the alarm exists and create it if it doesn't.
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 22]
```diff
@@ -0,0 +1,507 @@
...
+
+{% Aside 'important' %}
+
+All [Chrome API][doc-apis] event listeners and methods restart the service worker's 30 second termination timer. For more information, see the [Extension service worker lifecycle](tdb).
+
+{% endAside %}
+
+### Step 7: Communicate with other contexts {: #step-7 }
+
+Extensions use [content scripts][doc-content] to read and modify the content of the page. When a user visits a Chrome API reference page, the extension's content script will [send a message][doc-messages] to request the tip of the day from the service worker.
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:320]:
```suggestion
Extensions use [content scripts][doc-content] to read and modify the content of the page. When a user visits a Chrome API reference page, the extension's content script will update the page with the tip of the day. It [sends a message][doc-messages] to request the tip of the day from the service worker. 
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 23]
```diff
@@ -0,0 +1,507 @@
...
+
+The final step is to add a message handler to our service worker that sends a reply to the content script with the daily tip. 
+
+{% Label %}sw-api.js:{% endLabel %}
+
+```js
+...
+// Send tip to content script via messaging
+chrome.runtime.onMessage.addListener((message, sender, sendResponse) => {
+  if (message.greeting === 'tip') {
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:383]:
```suggestion
  if (message.type === 'tip') {
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 24]
```diff
@@ -0,0 +1,507 @@
...
+```
+
+Create a new content file. The following code sends a message to the service worker requesting the tip. Then, adds a button that will open a popover containing the extension tip. This code uses the new web platform [Popover API][popover-chrome-status] (see the [HTML spec][popover-html-spec] for more details). 
+
+{% Label %}content.js:{% endLabel %}
+
+```js
+(async () => {
+  // Sends a message to the service worker and receives a tip in response
+  const { tip } = await chrome.runtime.sendMessage({ greeting: 'tip' });
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:346]:
```suggestion
  const { tip } = await chrome.runtime.sendMessage({ type: 'tip' });
```

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 25]
```diff
@@ -0,0 +1,507 @@
...
+| Extension                        | What you will learn                                            |
+|----------------------------------|----------------------------------------------------------------|
+| [Reading time][tut-reading-time] | To insert an element on a specific set of pages automatically. |
+| [Tabs Manager][tut-tabs-manager] | To create a popup that manages browser tabs.                   |
+
+## Continue exploring
+
+To continue your extension service worker learning path, we recommend exploring the following articles:
+
+- TBD
```
Comment (sebastianbenz) [inline site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:468]: Have you already got a list of articles to link to here?

**Review by sebastianbenz — COMMENTED** [review]
Few comments, but looks great! 

I've noticed that the videos are quite small. Does dcc support opening inline videos in a lightbox?

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 26]
```diff
@@ -0,0 +1,507 @@
...
+layout: 'layouts/doc-post.njk'
+title: 'Handle events with service workers'
+seoTitle: 'Chrome extension service worker tutorial.'
+subhead: 'Tutorial that covers extension service worker concepts'
+description: 'Learn how to create and debug an extension service worker.'
+date: 2023-04-02
+# updated: 2022-06-13
+---
+
+## Overview {: #overview }
```
[the remark this answers is not in the document]
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:11]: Done! 👍 Also made the video larger

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 27]
```diff
@@ -0,0 +1,507 @@
...
+
+After 30 seconds you will see "service worker(inactive)" which means the service worker has terminated. Click on the "service worker(inactive)" hyperlink to inspect it. See example below.
+
+{% Video src="video/BhuKGJaIeLNPW9ehns59NfwqKxF2/D1XRaA6q4xn9Ylwe1u1N.mp4", width="800", height="314", autoplay="true", muted="true", loop="true" %}
+
+Did you notice that inspecting the service worker woke it up? That's right! Opening the service worker in the devtools will keep it active. To make sure that your extension behaves correctly when your service worker is terminated, remember to close the DevTools.
+
+Now let's break the extension to learn where to locate errors. One way to do this is to delete the ".js" from the `'./sw-omnibox.js'` import in the `service-worker.js` file. Chrome will be unable to register the service worker.
+
+Go back to chrome://extensions and refresh the extension. The following error will appear:
```
[the remark this answers is not in the document]
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:128]: Good idea... forgot about that! 😅 Thanks for pointing it out 👍

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md (hunk 17 above)
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md]: 👍 I added this to a fix links batch

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md (hunk 18 above)
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md]: Fixed 👍

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md (hunk 20 above)
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:271]: This is the convention we've been following for permissions and manifest keys

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md (hunk 16 above)
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:146]: Sounds good :)

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md (hunk 25 above)
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:474]: I was hoping to link to some Service worker "How-to's" or the explainer, but that is only one approach. 

We could also simply link our to general development articles. I'm open to suggestions.

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md (hunk 24 above)
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:352]: Thanks for this suggestion.  Since we already merged the sample code, I'd rather leave this as is for now. If we find a critical change to the source code, I will add this to the list.

File: site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md [hunk 28]
```diff
@@ -0,0 +1,516 @@
...
+
+See [Importing scripts](tbd) to learn about other ways to import multiple files in a service worker.
+
+{% Aside 'important' %}
+
+Remember to set `type.module` when using a modern module bundler framework, such as [CRXjs Vite plugin][crxjs-vite].
+
+{% endAside %}
+
+### _Optional: Debugging the service worker_ {: #step-3 }
```
[the remark this answers is not in the document]
Comment (rachelandrew) [reply site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md:115]: We don't use emphasis like this typically.

**Review by rachelandrew — APPROVED** [review]
LGTM with one small comment, I'm assuming all the TBD links will be fixed before merging.
